src/ErrorBoundary.ts

```diff
@@ -46,8 +56,30 @@ export class ErrorBoundary extends Component<
     }
   }
 
+  componentDidMount() {
+    window.addEventListener("error", handleSuppressLogging);
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener("error", handleSuppressLogging);
+    window.removeEventListener("error", handleRestoreLogging);
+  }
+
+  shouldComponentUpdate(
+    nextProps: ErrorBoundaryProps,
+    nextState: ErrorBoundaryState
+  ) {
+    if (!this.props.suppressLogging && nextState.didCatch) {
+      window.removeEventListener("error", handleSuppressLogging);
+      window.addEventListener("error", handleRestoreLogging);
+    }
```

Comment:
During my experiment on the implementation of the `suppressLogging`-prop-triggered behavior, I discovered that each error will reach the `window` object twice, and the error flows in the following way:
```mermaid
flowchart TD
subgraph ErrorBoundary
subgraph children
Child
end
Child --error thrown the first time--> window1["`window
(always suppresses error logging as long as at least one **ErrorBoundary** is mounted and the **_suppressLogging** key from the error is true)`" ] --> shouldComponentUpdate["`shouldComponentUpdate
(determines whether to log the error to the console according to the **suppressLogging** prop of the **ErrorBoundary**)`"] --error thrown the second time--> window2[window] --> componentDidCatch --> componentDidUpdate
end
```
Based on this observation, wrapping the side effects in `shouldComponentUpdate` is a compromised workaround. If the side effects are delayed to `componentDidUpdate`, it will be too late to catch the error at the second time it reaches the `window` object.